saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/util/Logging.scala

```diff
@@ -76,13 +77,40 @@ trait Logging {
     * format: ON
     */
   object loggerConfig {
+
     case class Logger(loggerName: String = org.slf4j.Logger.ROOT_LOGGER_NAME) {
-      private val logger: ch.qos.logback.classic.Logger =
-        LoggerFactory.getLogger(loggerName).asInstanceOf[ch.qos.logback.classic.Logger]
+      private val logger: ch.qos.logback.classic.Logger = getLogger()
+
+      private def getLogger(): ch.qos.logback.classic.Logger = {
+        if (LoggerFactory.getLogger(loggerName).isInstanceOf[ch.qos.logback.classic.Logger])
+          LoggerFactory.getLogger(loggerName).asInstanceOf[ch.qos.logback.classic.Logger]
+        else null
```

Comment:
if (x.isInstanceOf[B]) {
        val b = x.asInstanceOf[B];
        ...
    } else ...

can be written nicely as

    x match {
        case (b: B) => ...
        ...
    }